client/client-multi/client-c/CMakeLists.txt

```diff
@@ -275,7 +275,7 @@ else()
     add_definitions(-DKAA_DISABLE_FEATURE_NOTIFICATION)
 endif()
 
-if(KAA_ENCRYPTION)
+if(NOT DEFINED DISABLE_ENCRYPTION)
```

Comment:
use the way it was done for the extensions e.g. `option(KAA_ENCRYPTION "Enable encryption" ON)`